### PR TITLE
ISPN-15081 Patternfly 5 update

### PR DESCRIPTION
**Pull request by dpanshug**
https://issues.redhat.com/browse/ISPN-15081

**Summary of changes:**

1. Node and npm Update:
   - Updated Node.js version to v18.17.0 and npm to 9.6.7 in `github/pull_request.yaml` and `pom.xml`.

2. Patternfly Update:
   - Upgraded patternfly packages to latest versions (PF5) .

3. React Version Upgrade:
   - Transitioned to React 18.

4. Other Package Updates: Reviewed and updated all other packages for compatibility and enhancements, except the following:

    a. css-loader:
       - Requires adjustments to webpack configuration.
       - Another issue for it [ISPN-15145](https://issues.redhat.com/browse/ISPN-15145)

    b. react-router-dom:
       - Requires modifications to router config throughout the project.
       - Another issue for it [ISPN-15146](https://issues.redhat.com/browse/ISPN-15146)

    c. monaco-editor & react-monaco-editor:
       - Can't update due to PF5's reliance on outdated dependencies.

5. Separately modified background image on Welcome and About page.

### CODE REVIEW

**Comment by dpanshug** [comment]
Background image issue raise https://github.com/patternfly/patternfly/issues/5806

File: .cache_1b2yk2k
Comment (andyuk1986) [inline .cache_1b2yk2k:1]: Hey, I guess this file was committed by chance.

**Comment by dpanshug** [comment]
@karesti @andyuk1986 this PR is ready for final review

**Comment by dpanshug** [comment]
> Background image issue raise [patternfly/patternfly#5806](https://github.com/patternfly/patternfly/issues/5806)

Added custom CSS to solve this

**Comment by karesti** [comment]
@dpanshug added a commit. Popover was broken, I fixed that and a couple small unrelated changes introduced with the UX refactoring

**Review by karesti — APPROVED** [review]
looks good, I fixed the error with the Popover

Please double check the patternfly v5 css properties that were renamed

**Comment by dpanshug** [comment]
@andyuk1986 Can you please re-review this, some tests are failing.

File: src/app/Common/ConsoleBackground/ConsoleBackground.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.background-img {
```
Comment (karesti) [inline src/app/Common/ConsoleBackground/ConsoleBackground.css:1]: @dpanshug this does not work when you are in production built code. this gets overriden by the selector 

.pf-v5-c-background-image

we need to use the selectors in the components and check the variables associated

**Comment by karesti** [comment]
@dpanshug I'm on the review and some fixes

**Review by karesti — APPROVED** [review]
@dpanshug if you plan to update or do follow up PR that are not mandatory for this one, please, create JIRAS with the explanation and content of those changes to be done. If you can't do it, I or somebody else will take care of those changes.

**Comment by dpanshug** [comment]
> @dpanshug if you plan to update or do follow up PR that are not mandatory for this one, please, create JIRAS with the explanation and content of those changes to be done. If you can't do it, I or somebody else will take care of those changes.

Created JIRA's for those, [ISPN-15146](https://issues.redhat.com/browse/ISPN-15146) [ISPN-15145](https://issues.redhat.com/browse/ISPN-15145)

**Comment by karesti** [comment]
@andyuk1986 the test passed here so I merged, but it's failing in main

**Comment by andyuk1986** [comment]
Hey @karesti , I was on PTO. Will look at the tests in main.

**Pull request merged**